### ISSUE TITLE
Compilation problem

### ISSUE THREAD

**Issue by pemsley** [jenningsm42/Vulpes-Engine#5]
Hi Mark,

I tried to compile... but failed - I am not sure if this is my old compiler or your code.
(Apple LLVM version 6.1.0 (clang-602.0.49) (based on LLVM 3.6.0svn))

Thanks,

Paul.

> /Users/pe/Projects/vulpes/Vulpes-Engine/src/Scene.cpp:72:10:
> 
>  warning: enumeration value 'Other' not handled in
>       switch [-Wswitch]
>                 switch(type)
>                        ^
> /Users/pe/Projects/vulpes/Vulpes-Engine/src/ResourceLoader.cpp:142:81: error: cannot pass object of non-trivial type
>       'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >') through variadic
>       function; call will abort at runtime [-Wnon-pod-varargs]
>                         Logger::log("vul::ResourceLoader::loadTextureFromFile: Returned empty '%s'", path);
>                                                                                                      ^
> /Users/pe/Projects/vulpes/Vulpes-Engine/src/ResourceLoader.cpp:150:81: error: cannot pass object of non-trivial type
>       'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >') through variadic
>       function; call will abort at runtime [-Wnon-pod-varargs]
>                         Logger::log("vul::ResourceLoader::loadTextureFromFile: Unable to load '%s'", path);
>                                                                                                      ^
> 1 warning generated.
> 2 errors generated.
> make[2]: *** [CMakeFiles/vulpes.dir/src/ResourceLoader.cpp.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> [ 91%] Building CXX object CMakeFiles/vulpes.dir/src/VEMParser.cpp.o
> make[1]: *** [CMakeFiles/vulpes.dir/all] Error 2
> make: *** [all] Error 2
>

**Comment by jenningsm42** [comment]
Hi Paul,

Sorry for taking so long to reply. I think my compiler implicitly casts std::string to a C-style string there, so I didn't catch it. I'll make a PR shortly to fix that.

Also, there currently isn't any documentation on how to actually use the engine, so I'll upload a demo project to a new repo and link it in the README here (until I get around to actually writing docs - planned to be within the next few updates!).

Thanks for the heads up!
Mark

**Comment by pemsley** [comment]
On Sat, 13 Apr 2019 at 01:59, Mark Jennings <notifications@github.com>
wrote:

> Hi Paul,
>
> Sorry for taking so long to reply. I think my compiler implicitly casts
> std::string to a C-style string there, so I didn't catch it. I'll make a PR
> shortly to fix that.
>
> Also, there currently isn't any documentation on how to actually use the
> engine, so I'll upload a demo project to a new repo and link it in the
> README here (until I get around to actually writing docs - planned to be
> within the next few updates!).
>
>
Hi Mark,

I am the author of a molecular graphics application representing protein
models and cryo-EM reconstructions (that is in part what it does). I am
interested in improving its looks - and that's how I discovered your
Vulpes-engine.

This is how it looks now (with fixed pipeline graphics):

https://www2.mrc-lmb.cam.ac.uk/personal/pemsley/coot/web/screenshots/coot-200A-virus-page.html

https://www2.mrc-lmb.cam.ac.uk/personal/pemsley/coot/web/screenshots/coot-gleevec-page.html

The model primitives are 3D points with a color index (atoms) and lines
between 2 3D points (bonds). The density/mesh is just a list of lines
between points (or triangles).

I'd like to represent the density with some sort of glass effect, rather
than the triangular mesh.

Thanks for the (future) demo.

Regards,


Paul.

**Comment by jenningsm42** [comment]
Sorry again for the late reply - I just finished my final exams. 

Your application looks very interesting, but I don't think Vulpes Engine is a suitable fit for it (yet). In particular, it doesn't support transparent materials and it isn't entirely easy to implement it due to the rendering technique I chose.

If you're looking for a C++ library that can render in 3D, then you might want to check out Ogre3D or Irrlicht which are more feature-complete engines (there are many others which are easy to find, too). Godot also has C++ but I don't believe you import it as a library. If the language doesn't matter then Unity would be your best bet (especially considering the size of the community - you would get a lot of help).

I'll leave this issue up until I fix the code in the meantime. Let me know if you have any questions!

Mark

https://www.ogre3d.org/
http://irrlicht.sourceforge.net/
https://godotengine.org/
https://unity.com/
